yocto-update-strategies.md

```diff
@@ -0,0 +1,651 @@
+Yocto-based projects. Updating these components in a systematic manner reduces
+the risk of compatibility issues, ensures the project remains up to date, and
+allows for seamless integration of new functionalities and improvements.
+
+---
+# Performing update - build environment
+
+* Verify the compatibility of the build environment with the desired Yocto
+  Project release
+  - see `Build Host Packages` in Mega Manual
+  - use correct release
+* Prepare reproducible build environment
+  - ansible can be used to put together such an environment
+  - other possibility is to prepare containeraized image; e.g.
+    [kas](https://github.com/siemens/kas),
+    [imx-docker](https://github.com/nxp-imx/imx-docker),
+    [yocto-docker](https://github.com/3mdeb/yocto-docker)
+  - easier to distribute such an environment among colleagues
+* Review and update the configuration files (local.conf, bblayers.conf) to
+  reflect any changes or additions required for the update
+  - see release notes, to check if any changes need to be applied
+  - rethink the way build configuration files are stored
+
+???
+
+rethink the way build configuration files are stored - you may want to go back
+to the version before an update
+
+---
+# Performing update - external layers update
+
+.center[<img src="img/project-layers.svg" width="450px">]
+
+* Project consist of external and custom layers
+  - this way we follow the rule **Never modify the POKY layer. Never. Ever.**
+  - any customizations goes into `meta-custom_layer`: we have full control on
+    what and why was changed through `.bbappends`
+* For `external` we should pick Yocto release; specific commit hash
+  - task for that should be short, e.g. when using NXP BSP as base, we should
+    follow hashes used in given BSP release
+
+???
+
+Never modify the POKY layer. Never. Ever. - even expand it on other layers
+
+---
+# Performing update - custom layer update
+
+* Most of the effort of performing updates should be implemented in custom
+  layers
+* We can list some of the basic tasks carried out when updating Yocto, which
+  should be done in custom layers
+  - updating `LAYERSERIES_COMPAT` variable
+  - going through the bbappends files and verifying that they all still apply
+    (whether they depended on a revised version of a recipe)
+  - verification of custom patches for used packages, perhaps some of them were
+    added in the upstream
+  - syntax and variables verification
+  - path verification for the keywords `require` and `include`
+* Check the migration guides in case we encounter unexpected behavior
+  - there is one per every release
+  - we may found there usefull scripts, like one for override syntax change
+
+---
+# Build troubleshooting
+
+* We can split up every operating system into couple pieces
+.center[<img src="img/boot-process.svg" width="400px">]
+* Updating whole system, means that every part needs to work correctly (meaning,
```

Comment:
I would move these to speaker notes, so image can be bigger, and we can focus on your speech, not on reading slide.

You can leave some short bulleets if needed.

Reply:
bullets moved to notes https://github.com/3mdeb/conferences/pull/7/commits/f7e23ce4cb7a83fd0804e1f30ca618d34e5e63d2#diff-a1666090e402e433c955cc46797dc399867480b4d815574efd9f5d4181e24ac3R374